core/src/main/java/org/arquillian/cube/impl/client/container/ContainerConfigurationController.java

```diff
@@ -64,15 +64,16 @@ public void remapContainer(@Observes BeforeSetup event, CubeRegistry cubeRegistr
         Object newConfigurationInstance = configurationClass.newInstance();
 
         for (PropertyDescriptor configurationClassPortField : configurationClassPortFields) {
+            // Move it outside if condition below, if there is a group of containers.
+            int containerPort = getDefaultPortFromConfigurationInstance(newConfigurationInstance,
```

Comment:
I don't have enough knowledge on this code base to comment the correct approach.
However, generally speaking, you need the conditional to avoid a NPE.